src/test/scala/nl.knaw.dans.easy.deposit/servlets/RichZipInputStreamSpec.scala

```diff
@@ -15,36 +15,59 @@
  */
 package nl.knaw.dans.easy.deposit.servlets
 
-import java.io.ByteArrayInputStream
+import java.io.{ ByteArrayInputStream, FileInputStream, InputStream }
 import java.nio.charset.StandardCharsets
-import java.util.zip.ZipInputStream
 
-import better.files.File
 import nl.knaw.dans.easy.deposit.Errors.MalformedZipException
 import nl.knaw.dans.easy.deposit.TestSupportFixture
-import resource.managed
+import org.apache.commons.compress.archivers.zip.ZipArchiveInputStream
+import resource.{ ManagedResource, managed }
 
-import scala.util.{ Failure, Success }
+import scala.util.{ Failure, Success, Try }
 
 class RichZipInputStreamSpec extends TestSupportFixture {
 
+  private val stagingDir = testDir / "staging"
+
   override def beforeEach(): Unit = {
     super.beforeEach()
     clearTestDir()
   }
 
   "unzipPlainEntriesTo" should "report invalid content" in {
-    managed(new ZipInputStream(new ByteArrayInputStream("Lorem ipsum est".getBytes(StandardCharsets.UTF_8)))).apply(
-      _.unzipPlainEntriesTo(testDir / "dummy") should matchPattern {
-        case Failure(MalformedZipException("No entries found.")) =>
-      })
+    managedZipStream(new ByteArrayInputStream("Lorem ipsum est".getBytes(StandardCharsets.UTF_8))).apply(
+      unzip(_) shouldBe Failure(MalformedZipException("No entries found."))
+    )
   }
 
   it should "extract all files from the zip" in {
-    val stagingDir = testDir / "staging"
-    managed(File("src/test/resources/manual-test/Archive.zip").newZipInputStream).apply(
-      _.unzipPlainEntriesTo(stagingDir.createDirectories()) shouldBe Success(()))
+    managedZipStream(new FileInputStream("src/test/resources/manual-test/Archive.zip")).apply(
+      unzip(_) shouldBe Success(())
+    )
+    stagingDir.walk().map(_.name).toList should contain theSameElementsAs
+      List("staging", "login.html", "readme.md", "upload.html")
+  }
+
+  it should "not throw ZipException: only DEFLATED entries can have EXT descriptor" in {
+    managedZipStream(new FileInputStream("src/test/resources/manual-test/ruimtereis-bag.zip")).apply(
```

Comment:
Did not add a unit test test for the zip without directory entries because of the special license. Anybody any suggestions?